app/models/course/condition/achievement.rb

```diff
@@ -1,4 +1,5 @@
 class Course::Condition::Achievement < ActiveRecord::Base
-  acts_as :condition, class_name: Course::Condition.name
-  belongs_to :achievement, class_name: Course::Achievement.name
+  acts_as_condition
+  belongs_to :achievement,
```

Comment:
Because an achievement condition refers to an achievement (i.e. the user needs to unlock this achievement to satisfy the condition).